rasa/model.py

```diff
@@ -0,0 +1,274 @@
+import tempfile
+from typing import Text, Tuple, Union, Optional, List, Dict
+
+from rasa.constants import DEFAULT_MODELS_PATH
+
+# Type alias for the fingerprint
+Fingerprint = Dict[Text, Union[Text, List[Text]]]
+
+logger = logging.getLogger(__name__)
+
+FINGERPRINT_FILE_PATH = "fingerprint.json"
+
+FINGERPRINT_CONFIG_KEY = "config"
+FINGERPRINT_DOMAIN_KEY = "domain"
+FINGERPRINT_NLU_VERSION_KEY = "nlu_version"
+FINGERPRINT_CORE_VERSION_KEY = "core_version"
+FINGERPRINT_RASA_VERSION_KEY = "version"
+FINGERPRINT_STORIES_KEY = "stories"
+FINGERPRINT_NLU_DATA_KEY = "messages"
+
+
+def get_model(model_path: Text = DEFAULT_MODELS_PATH,
+              subdirectories: bool = False) -> Optional[Text]:
+    """Gets a model and unpacks it.
+
+    Args:
+        model_path: Path to the zipped model. If it's a directory, the latest
+                    trained model is returned.
+        subdirectories: If `True` it also returns the model subdirectories for
+                        Core and NLU, if `False` it only returns the path
+                        for the unpacked model directory.
+
+    Returns:
+        Path to the unpacked model.
+
+    """
+    if not model_path:
+        return None
+    elif os.path.isdir(model_path):
+        model_path = get_latest_model(model_path)
+
+    return unpack_model(model_path, subdirectories=subdirectories)
+
+
+def get_latest_model(model_path: Text = DEFAULT_MODELS_PATH) -> Optional[Text]:
+    """Gets the latest model from a path.
+
+    Args:
+        model_path: Path to a directory containing zipped models.
+
+    Returns:
+        Path to latest model in the given directory.
+
+    """
+    if not os.path.exists(model_path) or os.path.isfile(model_path):
+        model_path = os.path.dirname(model_path)
+
+    list_of_files = glob.glob(os.path.join(model_path, "*.tar.gz"))
+
+    if len(list_of_files) == 0:
+        return None
+
+    return max(list_of_files, key=os.path.getctime)
+
+
+def unpack_model(model_file: Text, working_directory: Text = None,
+                 subdirectories: bool = False
+                 ) -> Union[Text, Tuple[Text, Text, Text]]:
+    """Unpacks a zipped Rasa model.
+
+    Args:
+        model_file: Path to zipped model.
+        working_directory: Location where the model should be unpacked to.
+                           If `None` a tempory directory will be created.
+        subdirectories: If `True` it also returns the model subdirectories for
+                        Core and NLU, if `False` it only returns the path
+                        for the unpacked model directory.
+
+    Returns:
+        Path to unpacked Rasa model.
+
+    """
+    import tarfile
+
+    if working_directory is None:
+        working_directory = tempfile.mkdtemp()
+
+    tar = tarfile.open(model_file)
+    tar.extractall(working_directory)
+    tar.close()
+    logger.debug("Extracted model to '{}'.".format(working_directory))
+
+    model_directory = os.path.join(working_directory, "rasa_model")
```

Comment:
That was part of your prototype. No objections from my side to remove it.

Reply:
ok in that case I think we should remove it, don't see a benefit of having it

Reply:
I think you added this as this otherwise will be the name of the tar instead of `rasa_model` when you unpack it. Seems when I zip the files (without them being in another directory) it automatically adds everything in a directory with the tar file name, and zips that

Reply:
mhm I actually think that behaviour is not to bad - as right now if you unpack these models it will create folders named like `rasa_model 2`, `rasa_model 3`...

Reply:
done